### PR TITLE
DIV-8321 | Print facility staff should retrieve beneficiaries by phon…e and dob and can print certificates

### PR DESCRIPTION
**Pull request by varadeth**
@egov-joy @samarthgr please review

### CODE REVIEW

File: portal/src/components/PrintCertificate/PrintCertificate.js [hunk 1]
```diff
@@ -1,188 +1,56 @@
...
 function PrintCertificate() {
-    const [selectedReceipt, setSelectedReceipt] = React.useState([]);
-    const [selectedCertificate, setSelectedCertificate] = React.useState([]);
-    const [selectedProgram,setSelectedProgram] = React.useState();
-    const [programs, setPrograms] = useState([]);
-    const [tableData,setTableData] = React.useState(dummyTableData);
-    const receiptRef = React.useRef();
-    const certificateRef = React.useRef();
+    const today = new Date();
+    const [tableData,setTableData] = React.useState([]);
```
Comment (prasanna-egov) [inline portal/src/components/PrintCertificate/PrintCertificate.js:13]: Can be setState directly

**Review by prasanna-egov — COMMENTED** [review]
How do we handle downloading different certificate variants.

File: portal/src/components/PrintCertificate/PrintCertificate.js [hunk 2]
```diff
@@ -192,136 +60,63 @@ function PrintCertificate() {
...
-                        {selectedReceipt.length>0 ? <ReactToPrint
-                            trigger={() => <button className={styles['button']}>Print Receipt</button> }
-                            content={() => receiptRef.current}
-                        /> : ''}
-                        <div style={{ display: "none" }}>
-                            <CustomPrint ref={receiptRef} dataToPrint={selectedReceipt} title="Receipt"/>
-                        </div>
+                    <div className="col-sm-3 col-xs-3">
+                        <label className="custom-text-label required">Date of Birth</label>
+                        <CustomDateWidget options={options} onChange={(dob) => {setDob(dob); setDateValue(dob)}} value={dateValue} id="date-picker"></CustomDateWidget>
```
Comment (egov-joy) [inline portal/src/components/PrintCertificate/PrintCertificate.js:90]: should not need 2 state variables for DOB if the value being set is the same

**Comment by varadeth** [comment]
> How do we handle downloading different certificate variants.

currently it handles only vaccination certificates

File: portal/src/components/PrintCertificate/PrintCertificate.js (hunk 1 above)
Comment (varadeth) [reply portal/src/components/PrintCertificate/PrintCertificate.js:13]: ok

File: portal/src/components/PrintCertificate/PrintCertificate.js (hunk 2 above)
Comment (varadeth) [reply portal/src/components/PrintCertificate/PrintCertificate.js:90]: yes. that can be done

**Review by prasanna-egov — APPROVED** [review]

**Comment by prasanna-egov** [comment]
> > How do we handle downloading different certificate variants.
> 
> currently it handles only vaccination certificates

I meant EU, Who, smart health card.

**Review by egov-joy — APPROVED** [review]

**Pull request merged**